opentelemetry-sdk/src/resource/mod.rs

```diff
@@ -61,7 +63,29 @@ impl Default for Resource {
     }
 }
 
+impl From<ResourceBuilder> for Resource {
+    fn from(value: ResourceBuilder) -> Self {
+        value.build()
+    }
+}
+
 impl Resource {
+    /// Creates a Builder that allows you to configure multiple aspects of the Resource.
+    ///
+    /// If you want to start from a [Resource::default()] see [Resource::builder_default()].
+    ///
+    /// Starts with a [Resource::empty()].
+    pub fn builder() -> ResourceBuilder {
```

Comment:
I am still debating about this.. It maybe better to name this builder_empty, to make it very obvious that this starts with an empty Resource, and not with OTel SDK defaults?